packages/shared/src/components/modals/bookmark/BookmarkFolderModal.tsx

```diff
@@ -0,0 +1,221 @@
+        type="submit"
+        className="tablet:flex-grow"
+        disabled={!valid}
+        variant={ButtonVariant.Primary}
+      >
+        {!isPlus && folderCount > 0 ? (
+          <>
+            <DevPlusIcon /> Upgrade to plus
+          </>
+        ) : (
+          'Create folder'
+        )}
+      </Button>
+    </div>
+  );
+};
+
+const BookmarkFolderModal = ({
+  folder,
+  onSubmit,
+  folderCount = 0,
+  ...rest
+}: BookmarkFolderModalProps): ReactElement => {
+  const { isPlus } = usePlusSubscription();
+  const [valid, setValid] = useState(false);
+  const isMobile = useViewSize(ViewSize.MobileL);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // TODO: Dummy component for now. Implement request later.
+    // const formJson = formToJson(e.currentTarget);
+  };
+
+  return (
+    <Modal kind={Modal.Kind.FlexibleCenter} size={Modal.Size.Small} {...rest}>
+      <form onSubmit={handleSubmit}>
+        {isMobile && (
+          <FormActions
+            valid={valid}
+            folderCount={folderCount}
+            isPlus={isPlus}
+            isMobile={isMobile}
+          />
+        )}
+        <ModalHeader
+          showCloseButton={!isMobile}
+          className={classNames('gap-2', isMobile && '!border-b-0 pb-0')}
+        >
+          <ModalHeader.Title>New Folder</ModalHeader.Title>
+          <Typography
+            tag={TypographyTag.Span}
+            type={TypographyType.Caption1}
+            className="flex items-center rounded-4 bg-action-plus-float px-1 font-bold"
+            color={TypographyColor.Plus}
+          >
+            <DevPlusIcon size={IconSize.Size16} />
+            Plus
+          </Typography>
+        </ModalHeader>
+        <Modal.Body className="flex flex-col gap-5 tablet:gap-4">
+          {!isPlus && <PlusCTA folderCount={folderCount} />}
+          <TextField
+            maxLength={50}
+            label="Give your folder a name..."
+            name="name"
+            inputId="newFolder"
+            onChange={(e) => setValid(e.target.value.length > 0)}
+            value={folder?.name || ''}
+          />
+          <Typography className="font-bold" type={TypographyType.Body}>
+            Choose an icon
+          </Typography>
+          <ul className="flex flex-wrap justify-evenly gap-4">
+            {emojiOptions.map((emoji) => (
+              <li key={emoji}>
+                <label
+                  htmlFor={emoji || 'noicon'}
+                  className="flex cursor-pointer items-center"
+                >
+                  <input
+                    type="radio"
+                    name="icon"
+                    id={emoji || 'noicon'}
+                    value={emoji}
+                    className="peer hidden"
+                  />
+                  <span className="flex h-12 w-12 select-none items-center justify-center rounded-14 border-2 border-transparent bg-overlay-float-salt peer-checked:border-surface-focus">
```

Comment:
What does peer-checked do?
I'm actually thinking should we just render this as Buttons and simply set state on selected on?

If not, I'm not sure the colors match here, as they existing button colors

```suggestion
                  <span className="flex size-12 select-none items-center justify-center rounded-14 border-2 border-transparent bg-overlay-float-salt peer-checked:border-surface-focus">
```

Reply:
The colors do match. Peer is a utility class for Tailwind that lets you alter css based on peers. Here, when the checkbox is selected, we add the blue border around the selected box. Quite need functionality, actually! 😄 

[Docs](https://tailwindcss.com/docs/hover-focus-and-other-states)

As for using buttons, I'm not sure I agree. I think this is exactly what radio groups are for. What would be the benefit of converting to buttons?